doc/macsec-gearbox/Copper_PHY_Attributes_Proposal.md

```diff
@@ -0,0 +1,305 @@
+SAI_PORT_ATTR_MODULE_TYPE allows user to force various module/transceivers types when fiber interface is active. Supported fiber interfaces are 1000BASE-X, SGMII-SLAVE, and 100FX.
+
+```
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_MODULE_TYPE
+ * Used for configuring Fiber module type
+ */
+typedef enum _sai_port_module_type_t
+{
+    /** Module Type Fiber */
+    SAI_PORT_MODULE_TYPE_1000BASE_X,
+
+    /** Module Type 100FX */
+    SAI_PORT_MODULE_TYPE_100FX,
+
+    /** Module Type SGMII-Slave */
+    SAI_PORT_MODULE_TYPE_SGMII_SLAVE,
+} sai_port_module_type_t;
+
+/**
+ * @brief Configure Fiber module type
+ *
+ * @type sai_port_module_type_t
+ * @flags CREATE_AND_SET
+ * @default SAI_PORT_MODULE_TYPE_1000BASE_X
+ * @validonly SAI_PORT_ATTR_MEDIA_TYPE == SAI_PORT_MEDIA_TYPE_FIBER and SAI_PORT_ATTR_SPEED == 1000
+ */
+SAI_PORT_ATTR_MODULE_TYPE,
+
+```
+
+## Dual Media Support
+
+Dual media support allows user to configure Copper and Fiber supported External PHYs. When a setup has both Copper and Fiber media types but only one media type can be active at any given time. SAI_PORT_ATTR_DUAL_MEDIA attribute can be used to either use Copper only, Fiber only, Copper preferred, or Fiber preferred ports. SAI_PORT_DUAL_MEDIA_NONE will be used if the External PHY does not support dual media configuration.
+
+```
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_DUAL_MEDIA
+ * Used to configure media type for dual media supported PHY
+ */
+typedef enum _sai_port_dual_media_t
+{
+    /**  Dual media not supported */
+    SAI_PORT_DUAL_MEDIA_NONE,
+
+    /**  Force Copper mode, Fiber is inactive/disabled */
+    SAI_PORT_DUAL_MEDIA_COPPER_ONLY,
+
+    /**  Force Fiber mode, Copper is inactive/disabled */
+    SAI_PORT_DUAL_MEDIA_FIBER_ONLY,
+
+    /**  Both Copper and Fiber supported, but Copper preferred */
+    SAI_PORT_DUAL_MEDIA_COPPER_PREFERRED,
+
+    /**  Both Copper and Fiber supported, but Fiber preferred */
+    SAI_PORT_DUAL_MEDIA_FIBER_PREFERRED
+} sai_port_dual_media_t;
+
+/**
+ * @brief Configure media types for dual media supported PHY
+ *
+ * @type sai_port_dual_media_t
+ * @flags CREATE_AND_SET
+ * @default SAI_PORT_DUAL_MEDIA_NONE
+ */
+SAI_PORT_ATTR_DUAL_MEDIA,
+
+```
+## Cable Diagnostics Support
+
+Cable diagnostic feature allows user to query cable pair status and cable length based on the cable type provided by the user. This is important diagnostic feature to make sure that cable connected is working properly.
+
+Cable Pair State
+
+Copper cable has multiple pairs and each pair can be in different state as mentioned in the enum below:
+
+```
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_CABLE_PAIR_STATE
+ * Copper cable pair states
+ */
+typedef enum _sai_port_cable_pair_state_t
+{
+    /**  Cable state no faults */
+    SAI_PORT_CABLE_PAIR_STATE_OK,
+
+    /**  Cable state open */
+    SAI_PORT_CABLE_PAIR_STATE_OPEN,
+
+    /**  Cable state short */
+    SAI_PORT_CABLE_PAIR_STATE_SHORT,
+
+    /**  Cable state cross talk */
```

Comment:
Is it expected that OPENSHORT will not be available in the PAI?
i.e.
#define BCM_PORT_CABLE_STATE_OK         _SHR_PORT_CABLE_STATE_OK
#define BCM_PORT_CABLE_STATE_OPEN       _SHR_PORT_CABLE_STATE_OPEN
#define BCM_PORT_CABLE_STATE_SHORT      _SHR_PORT_CABLE_STATE_SHORT
#define BCM_PORT_CABLE_STATE_OPENSHORT  _SHR_PORT_CABLE_STATE_OPENSHORT
#define BCM_PORT_CABLE_STATE_CROSSTALK  _SHR_PORT_CABLE_STATE_CROSSTALK
#define BCM_PORT_CABLE_STATE_UNKNOWN    _SHR_PORT_CABLE_STATE_UNKNOWN
#define BCM_PORT_CABLE_STATE_COUNT      _SHR_PORT_CABLE_STATE_COUNT

/* bcm_port_cable_diag_t */
typedef _shr_port_cable_diag_t bcm_port_cable_diag_t;

Reply:
No. We don't support OPENSHORT. You can extend the enum if you want to support it once this is approved.